system_tests/pubsub.py

```diff
@@ -251,13 +262,15 @@ def test_fetch_delete_subscription_w_deleted_topic(self):
         subscription.create()
         topic.delete()
 
-        all_subs = []
-        token = None
-        while True:
-            subs, token = Config.CLIENT.list_subscriptions(page_token=token)
-            all_subs.extend(subs)
-            if token is None:
-                break
+        def _fetch():
+            return list(MethodIterator(Config.CLIENT.list_subscriptions))
+
+        def _found_orphan(result):
+            return any(subscription for subscription in result
+                       if subscription.name == ORPHANED)
```

Comment:
`len(result)` may be greater than one, so we have to filter somehow.  To me, that expression maps naturally onto "Are there any subscriptions whose name is the expected one?"  Converting the generator expression to a list comprehension and then taking its length doesn't seem clearer to me.  Compare the following to the version above:

```python
        def _found_orphan(result):
            return len([subscription for subscription in result
                        if subscription.name == ORPHANED]) == 1
```